File: encryption/src/main/java/com/hitrontech/hitronencryption/EncryptionImpl.java
package com.hitrontech.hitronencryption;

import android.content.Context;

public interface EncryptionImpl {

    String base64Encoder(String key, String text);

    String base64Encoder(Context context, String key, String text);

    String base64EncoderByAppId(String applicationId, String key, String text);

    String base64Decoder(String key, String encoderStr);

    String base64Decoder(Context context, String key, String encoderStr);

    String base64DecoderByAppId(String applicationId, String key, String encoderStr);

    String md5File(String filePath);

    String md5Str(String text);

}

File: encryption/src/main/java/com/hitrontech/hitronencryption/EncryptionImpl.kt
package com.hitrontech.hitronencryption

import android.content.Context

interface EncryptionImpl {

    fun base64Encoder(key: String, text: String): String

    fun base64Encoder(context: Context?, key: String, text: String): String

    fun base64EncoderByAppId(applicationId: String?, key: String, text: String): String

    fun base64Decoder(key: String, encoderStr: String): String?

    fun base64Decoder(context: Context?, key: String, encoderStr: String): String?

    fun base64DecoderByAppId(applicationId: String?, key: String, encoderStr: String): String?

    fun md5File(filePath: String): String?

    fun md5Str(text: String): String?

}
